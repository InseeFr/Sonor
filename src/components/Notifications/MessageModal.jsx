import React from 'react';
import './Notifications.css';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import D from '../../i18n';
import 'react-bootstrap-typeahead/css/Typeahead.css';

function MessageModal({ messageView, setMessageView }) {
  return (
    <Modal
      show={messageView !== null}
      onHide={() => setMessageView(null)}
    >
      { messageView ? (
        <>
          <Modal.Header closeButton>
            <Modal.Title>
              <div>
                {`${new Date(messageView.date).toLocaleDateString().slice(0, 5)} ${(`0${new Date(messageView.date).getHours()}`).slice(-2)}h${(`0${new Date(messageView.date).getMinutes()}`).slice(-2)}`}
              </div>
            </Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <div>
              <b>
                {`${D.from}: `}
              </b>
              {messageView.sender}
            </div>
            <hr />
            <div>
              <b>
                {`${D.to}: `}
              </b>
              {messageView.typedRecipients.map((r) => `@${r.id}${r.label ? ` (${r.label.length < 20 ? r.label : `${r.label.slice(0, 20)}...`})` : ''}`)
                .join(', ')}
            </div>
            <hr />
            {
            messageView
              ? messageView.text.split(/\n\r?/g)
                .map((line, index) => (index ? (
                  <>
                    <br />
                    {line}
                  </>
                ) : line))
              : ''
            }
          </Modal.Body>
          <Modal.Footer>
            <Button
              data-testid="close-preferences-button"
              onClick={() => setMessageView(null)}
            >
              OK
            </Button>
          </Modal.Footer>
        </>
      ) : null}
    </Modal>
  );
}

export default MessageModal;
