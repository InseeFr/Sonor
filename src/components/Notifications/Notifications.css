
#Container{
	position: absolute;
	width: 100%;
	height: 100%;
}
#mailCard{
  /* position: absolute; */
  /* width: 340px ; */
	height: 96%;
	padding:10px;
  /* margin: 20px 3% 50px; */
  margin-top: 10px;
}

#message{
	height: 200px;
}

#sendButton{
	margin-top: 10px;
	margin-left: auto;
}

#recipientWrapper {
  position: relative
}

#badgeWrapper {
  position: absolute;
  top: 0;
}

#recipients{
  width: 100%;
}

.badge {
  margin-left: 5px;
}

.lastDiv {
  display: inline-block;
}


.textwidth {
  position: absolute;
  visibility: hidden;
  height: auto;
  width: auto;
}

textarea {
  resize: none;
}

#mainCard {
  position: relative;
  display:inline-block;
  left: 5%;
  right:5%;
  width:90%;
  height: 75%;
}

.card{
  box-shadow: 0 .5rem 1rem rgba(0,0,0,.15)!important;
}

#NotificationHistory{
	height: 96%;
	padding:10px;
  /* margin: 20px 3% 50px; */
  margin-top: 10px;
}

#mailCardCol{
  max-width: 50%;
  display: inline-block;
  width: 50%;
}

#messageHistory{
  text-overflow: ellipsis;
}

td {
  height: 50px;
}

.time{
  width: 65px; 
}

.sender{
  width: 50px; 
}

#historyCol{
  display: inline-block;
  width: 50%;
  height: 100%;
}

#notificationReturnButton{
  margin: 0 0 10px 15px
}

.HistoryLine{
  cursor: pointer;
}