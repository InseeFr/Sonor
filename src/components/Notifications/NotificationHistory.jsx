import React, { useState } from 'react';
import './Notifications.css';
import Card from 'react-bootstrap/Card';
import Table from 'react-bootstrap/Table';
import PaginationNav from '../PaginationNav/PaginationNav';
import D from '../../i18n';

function NotificationHistory({ setMessageView, messageHistory }) {
  const [pagination, setPagination] = useState({ size: 10, page: 1 });

  return (
    <Card id="NotificationHistory">
      <Card.Title>{D.history}</Card.Title>
      <Table id="messageHistory" className="CustomTable" bordered striped hover responsive size="sm">
        <tbody>
          {messageHistory.slice(
            (pagination.page - 1) * pagination.size,
            Math.min(pagination.page * pagination.size, messageHistory.length),
          ).map((msg) => (
            <tr
              className="HistoryLine"
              onClick={() => {
                setMessageView(msg);
              }}
            >
              <td className="time">{`${new Date(msg.date).toLocaleDateString().slice(0, 5)} ${(`0${new Date(msg.date).getHours()}`).slice(-2)}h${(`0${new Date(msg.date).getMinutes()}`).slice(-2)}`}</td>
              <td className="sender">
                <b>
                  {`${D.from}: `}
                </b>
                {msg.sender}
              </td>
              <td className="recipients">
                <b>
                  {`${D.to}: `}
                </b>
                {msg.typedRecipients.map((r) => `@${r.id}${r.label ? ` (${r.label.length < 20 ? r.label : `${r.label.slice(0, 20)}...`})` : ''}`)
                  .join(', ')}
              </td>
              <td>
                {msg.text}
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
      <div className="tableOptionsWrapper">
        <PaginationNav.PageSelector
          pagination={pagination}
          updateFunc={(newPagination) => { setPagination(newPagination); }}
          numberOfItems={messageHistory.length}
        />
      </div>
    </Card>
  );
}

export default NotificationHistory;
