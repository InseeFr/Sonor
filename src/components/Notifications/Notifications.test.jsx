import React from 'react';
import { act } from 'vitest';
import { render, screen, fireEvent, cleanup, waitForElement, wait } from '@testing-library/react';
import { Router, Route, Switch } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import DataFormatter from '../../utils/DataFormatter';
import Notifications from './Notifications';
import mocks from '../../tests/mocks';
import userEvent from '@testing-library/user-event';

import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import waitUntil from 'async-wait-until';

configure({ adapter: new Adapter() });

const history = createMemoryHistory();

const toLocaleDateString = Date.prototype.toLocaleString;
const getHours = Date.prototype.getUTCHours;
const getMinutes = Date.prototype.getUTCMinutes;
Date.prototype.toLocaleDateString = function () {
  return toLocaleDateString.call(this, 'en-EN', {
    timeZone: 'UTC',
    year: 'numeric',
    month: 'numeric',
    day: 'numeric',
  });
};
Date.prototype.getHours = function () {
  return getHours.call(this);
};
Date.prototype.getMinutes = function () {
  return getMinutes.call(this);
};

global.document.createRange = () => ({
  setStart: () => {},
  setEnd: () => {},
  commonAncestorContainer: {
    nodeName: 'BODY',
    ownerDocument: document,
  },
});

const OriginalDate = global.Date;
jest
  .spyOn(global, 'Date')
  .mockImplementation(a =>
    a ? new OriginalDate(a) : new OriginalDate('2020-08-20T11:01:58.135Z')
  );
Date.now = jest.fn(() => 1597916474000);

beforeEach(() => {
  history.push('/notifications');
});

afterEach(cleanup);

jest.mock('../../utils/DataFormatter');

const survey = mocks.surveyVqs;
const resp = mocks.listSU;
const { messageHistory, verifyNameResp } = mocks;

const TestingRouter = ({ ComponentWithRedirection }) => (
  <Router history={history}>
    <Switch>
      <Route path="/notifications" component={() => <ComponentWithRedirection />} />
      <Route
        path="*"
        render={routeProps => (
          <div>
            <div data-testid="Redirect-url">
              {JSON.stringify(routeProps.history.location.pathname)}
            </div>
            <div data-testid="Redirect-survey">
              {!routeProps.history.location ||
                !routeProps.history.location.survey ||
                JSON.stringify(routeProps.history.location.survey)}
            </div>
          </div>
        )}
      />
    </Switch>
  </Router>
);
const postMessageMock = jest.fn();
const verifyNameMock = jest.fn((q, c) => c(verifyNameResp));
DataFormatter.mockImplementation(() => ({
  getMessageHistory: c => c(messageHistory),
  verifyName: verifyNameMock,
  postMessage: postMessageMock,
}));

const mockDataRetreiver = new DataFormatter();

it('Component is correctly displayed', async () => {
  const component = render(
    <Router history={history}>
      <Notifications user={{ id: 'test' }} dataRetreiver={mockDataRetreiver} />
    </Router>
  );

  // Should match snapshot (rows displayed)
  expect(component).toMatchSnapshot();
});

it('Change page', async () => {
  const component = render(
    <Router history={history}>
      <Notifications user={{ id: 'test' }} dataRetreiver={mockDataRetreiver} />
    </Router>
  );

  screen.getByTestId('pagination-nav').lastChild.firstChild.click();

  // Should match snapshot (rows displayed have changed)
  expect(component).toMatchSnapshot();
});

it('Send message', async () => {
  // Using enzyme library for this test

  const wrapper = mount(
    <Router history={history}>
      <Notifications user={{ id: 'test' }} dataRetreiver={mockDataRetreiver} />
    </Router>
  );
  // Triggering the suggestion search by simulating input in typeahead component
  const input = wrapper.find('.rbt-input-main');
  const event = {
    defaultPrevented: false,
    persist: () => {},
    preventDefault: () => {},
  };
  const inputEvent = {
    defaultPrevented: false,
    persist: () => {},
    preventDefault: () => {},
    currentTarget: { value: 'm' },
  };
  input.prop('onChange')(inputEvent);

  // VerifyName should have been called
  expect(verifyNameMock).toHaveBeenLastCalledWith('m', expect.anything());

  // Writting a message in the textArea
  const messageInput = wrapper.find('#message');
  messageInput.simulate('change', { target: { value: 'a message' } });

  // focus is needed to trigger the suggestion overlay
  input.simulate('focus', event);
  // Choosing the first suggestion by clicking on it
  const option = wrapper.find('#async-typehead-item-0.dropdown-item');
  option.simulate('click');

  const sendbtn = wrapper.find('#sendButton').first();
  sendbtn.simulate('click');

  expect(postMessageMock).toHaveBeenLastCalledWith(
    { recipients: ['INTW1'], sender: 'test', text: 'a message' },
    expect.anything()
  );
});
