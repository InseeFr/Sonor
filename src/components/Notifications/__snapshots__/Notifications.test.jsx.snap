// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="Container"
      >
        <a
          class="ButtonLink ReturnButtonLink"
          href="/"
        >
          <button
            class="ReturnButton btn btn-primary"
            data-testid="return-button"
            id="notificationReturnButton"
            type="button"
          >
            Back
          </button>
        </a>
        <div
          class="card"
          id="mainCard"
        >
          <div
            class="col"
            id="mailCardCol"
          >
            <div
              class="card"
              id="mailCard"
            >
              <div
                class="card-title h5"
              >
                Notify interviewers
              </div>
              <span>
                Recipients : 
              </span>
              <div
                id="recipientWrapper"
              >
                <div
                  class="rbt"
                  style="outline: none; position: relative;"
                  tabindex="-1"
                >
                  <div
                    class="rbt-input-multi form-control rbt-input form-control-sm"
                    tabindex="-1"
                  >
                    <div
                      class="rbt-input-wrapper"
                    >
                      <div
                        style="display: flex; flex: 1; height: 100%; position: relative;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-haspopup="listbox"
                          autocomplete="off"
                          class="rbt-input-main"
                          style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; width: 100%; z-index: 1;"
                          type="text"
                          value=""
                        />
                        <input
                          aria-hidden="true"
                          class="rbt-input-hint"
                          readonly=""
                          style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 0px 0px 0px 0px; line-height: normal; padding: 0px 0px 0px 0px;"
                          tabindex="-1"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span>
                Message : 
              </span>
              <textarea
                id="message"
                type="textarea"
              />
              <button
                class="btn btn-primary"
                id="sendButton"
                type="button"
              >
                <i
                  class="fa fa-fw fa-paper-plane"
                />
                 
                Send
              </button>
            </div>
          </div>
          <div
            class="col"
            id="historyCol"
          >
            <div
              class="card"
              id="NotificationHistory"
            >
              <div
                class="card-title h5"
              >
                History
              </div>
              <div
                class="table-responsive"
              >
                <table
                  class="CustomTable table table-sm table-striped table-bordered table-hover"
                  id="messageHistory"
                >
                  <tbody>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h32
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW3 (Gerald Edwards)
                      </td>
                      <td>
                        test 7
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="tableOptionsWrapper"
              >
                <ul
                  class="paginationNav pagination pagination-sm"
                  data-testid="pagination-nav"
                >
                  <li
                    class="page-item"
                  >
                    <a
                      class="page-link"
                      href="#"
                      role="button"
                    >
                      1
                    </a>
                  </li>
                  <li
                    class="page-item active"
                  >
                    <span
                      class="page-link"
                    >
                      2
                      <span
                        class="sr-only"
                      >
                        (current)
                      </span>
                    </span>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="textwidth"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="Container"
    >
      <a
        class="ButtonLink ReturnButtonLink"
        href="/"
      >
        <button
          class="ReturnButton btn btn-primary"
          data-testid="return-button"
          id="notificationReturnButton"
          type="button"
        >
          Back
        </button>
      </a>
      <div
        class="card"
        id="mainCard"
      >
        <div
          class="col"
          id="mailCardCol"
        >
          <div
            class="card"
            id="mailCard"
          >
            <div
              class="card-title h5"
            >
              Notify interviewers
            </div>
            <span>
              Recipients : 
            </span>
            <div
              id="recipientWrapper"
            >
              <div
                class="rbt"
                style="outline: none; position: relative;"
                tabindex="-1"
              >
                <div
                  class="rbt-input-multi form-control rbt-input form-control-sm"
                  tabindex="-1"
                >
                  <div
                    class="rbt-input-wrapper"
                  >
                    <div
                      style="display: flex; flex: 1; height: 100%; position: relative;"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-haspopup="listbox"
                        autocomplete="off"
                        class="rbt-input-main"
                        style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; width: 100%; z-index: 1;"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden="true"
                        class="rbt-input-hint"
                        readonly=""
                        style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 0px 0px 0px 0px; line-height: normal; padding: 0px 0px 0px 0px;"
                        tabindex="-1"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <span>
              Message : 
            </span>
            <textarea
              id="message"
              type="textarea"
            />
            <button
              class="btn btn-primary"
              id="sendButton"
              type="button"
            >
              <i
                class="fa fa-fw fa-paper-plane"
              />
               
              Send
            </button>
          </div>
        </div>
        <div
          class="col"
          id="historyCol"
        >
          <div
            class="card"
            id="NotificationHistory"
          >
            <div
              class="card-title h5"
            >
              History
            </div>
            <div
              class="table-responsive"
            >
              <table
                class="CustomTable table table-sm table-striped table-bordered table-hover"
                id="messageHistory"
              >
                <tbody>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h32
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW3 (Gerald Edwards)
                    </td>
                    <td>
                      test 7
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="tableOptionsWrapper"
            >
              <ul
                class="paginationNav pagination pagination-sm"
                data-testid="pagination-nav"
              >
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                    role="button"
                  >
                    1
                  </a>
                </li>
                <li
                  class="page-item active"
                >
                  <span
                    class="page-link"
                  >
                    2
                    <span
                      class="sr-only"
                    >
                      (current)
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="textwidth"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Component is correctly displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="Container"
      >
        <a
          class="ButtonLink ReturnButtonLink"
          href="/"
        >
          <button
            class="ReturnButton btn btn-primary"
            data-testid="return-button"
            id="notificationReturnButton"
            type="button"
          >
            Back
          </button>
        </a>
        <div
          class="card"
          id="mainCard"
        >
          <div
            class="col"
            id="mailCardCol"
          >
            <div
              class="card"
              id="mailCard"
            >
              <div
                class="card-title h5"
              >
                Notify interviewers
              </div>
              <span>
                Recipients : 
              </span>
              <div
                id="recipientWrapper"
              >
                <div
                  class="rbt"
                  style="outline: none; position: relative;"
                  tabindex="-1"
                >
                  <div
                    class="rbt-input-multi form-control rbt-input form-control-sm"
                    tabindex="-1"
                  >
                    <div
                      class="rbt-input-wrapper"
                    >
                      <div
                        style="display: flex; flex: 1; height: 100%; position: relative;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-haspopup="listbox"
                          autocomplete="off"
                          class="rbt-input-main"
                          style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; width: 100%; z-index: 1;"
                          type="text"
                          value=""
                        />
                        <input
                          aria-hidden="true"
                          class="rbt-input-hint"
                          readonly=""
                          style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 0px 0px 0px 0px; line-height: normal; padding: 0px 0px 0px 0px;"
                          tabindex="-1"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <span>
                Message : 
              </span>
              <textarea
                id="message"
                type="textarea"
              />
              <button
                class="btn btn-primary"
                id="sendButton"
                type="button"
              >
                <i
                  class="fa fa-fw fa-paper-plane"
                />
                 
                Send
              </button>
            </div>
          </div>
          <div
            class="col"
            id="historyCol"
          >
            <div
              class="card"
              id="NotificationHistory"
            >
              <div
                class="card-title h5"
              >
                History
              </div>
              <div
                class="table-responsive"
              >
                <table
                  class="CustomTable table table-sm table-striped table-bordered table-hover"
                  id="messageHistory"
                >
                  <tbody>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h32
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW3 (Gerald Edwards)
                      </td>
                      <td>
                        test 7
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h31
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @vqs2021x00 (Everyday life and he...)
                      </td>
                      <td>
                        test 6
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h31
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW3 (Gerald Edwards)
                      </td>
                      <td>
                        test 5
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h31
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW4 (Melody Grant)
                      </td>
                      <td>
                        test 4
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h31
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW2 (Carlton Campbell)
                      </td>
                      <td>
                        test 3
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h31
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW1 (Margie Lucas)
                      </td>
                      <td>
                        test 2
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h30
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW1 (Margie Lucas)
                      </td>
                      <td>
                        test
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h30
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @INTW1 (Margie Lucas)
                      </td>
                      <td>
                        test
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/28/ 08h30
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @OU-NATIONAL (National organizatio...)
                      </td>
                      <td>
                        message 3
                      </td>
                    </tr>
                    <tr
                      class="HistoryLine"
                    >
                      <td
                        class="time"
                      >
                        4/27/ 16h27
                      </td>
                      <td
                        class="recipients"
                      >
                        <b>
                          To: 
                        </b>
                        @simpsons2020x00 (Survey on the Simpso...)
                      </td>
                      <td>
                        message de test
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="tableOptionsWrapper"
              >
                <ul
                  class="paginationNav pagination pagination-sm"
                  data-testid="pagination-nav"
                >
                  <li
                    class="page-item active"
                  >
                    <span
                      class="page-link"
                    >
                      1
                      <span
                        class="sr-only"
                      >
                        (current)
                      </span>
                    </span>
                  </li>
                  <li
                    class="page-item"
                  >
                    <a
                      class="page-link"
                      href="#"
                      role="button"
                    >
                      2
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="textwidth"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="Container"
    >
      <a
        class="ButtonLink ReturnButtonLink"
        href="/"
      >
        <button
          class="ReturnButton btn btn-primary"
          data-testid="return-button"
          id="notificationReturnButton"
          type="button"
        >
          Back
        </button>
      </a>
      <div
        class="card"
        id="mainCard"
      >
        <div
          class="col"
          id="mailCardCol"
        >
          <div
            class="card"
            id="mailCard"
          >
            <div
              class="card-title h5"
            >
              Notify interviewers
            </div>
            <span>
              Recipients : 
            </span>
            <div
              id="recipientWrapper"
            >
              <div
                class="rbt"
                style="outline: none; position: relative;"
                tabindex="-1"
              >
                <div
                  class="rbt-input-multi form-control rbt-input form-control-sm"
                  tabindex="-1"
                >
                  <div
                    class="rbt-input-wrapper"
                  >
                    <div
                      style="display: flex; flex: 1; height: 100%; position: relative;"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-haspopup="listbox"
                        autocomplete="off"
                        class="rbt-input-main"
                        style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; width: 100%; z-index: 1;"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden="true"
                        class="rbt-input-hint"
                        readonly=""
                        style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.54); left: 0px; pointer-events: none; position: absolute; top: 0px; width: 100%; border-style: inset inset inset inset; border-width: 0px 0px 0px 0px; line-height: normal; padding: 0px 0px 0px 0px;"
                        tabindex="-1"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <span>
              Message : 
            </span>
            <textarea
              id="message"
              type="textarea"
            />
            <button
              class="btn btn-primary"
              id="sendButton"
              type="button"
            >
              <i
                class="fa fa-fw fa-paper-plane"
              />
               
              Send
            </button>
          </div>
        </div>
        <div
          class="col"
          id="historyCol"
        >
          <div
            class="card"
            id="NotificationHistory"
          >
            <div
              class="card-title h5"
            >
              History
            </div>
            <div
              class="table-responsive"
            >
              <table
                class="CustomTable table table-sm table-striped table-bordered table-hover"
                id="messageHistory"
              >
                <tbody>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h32
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW3 (Gerald Edwards)
                    </td>
                    <td>
                      test 7
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h31
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @vqs2021x00 (Everyday life and he...)
                    </td>
                    <td>
                      test 6
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h31
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW3 (Gerald Edwards)
                    </td>
                    <td>
                      test 5
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h31
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW4 (Melody Grant)
                    </td>
                    <td>
                      test 4
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h31
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW2 (Carlton Campbell)
                    </td>
                    <td>
                      test 3
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h31
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW1 (Margie Lucas)
                    </td>
                    <td>
                      test 2
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h30
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW1 (Margie Lucas)
                    </td>
                    <td>
                      test
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h30
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @INTW1 (Margie Lucas)
                    </td>
                    <td>
                      test
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/28/ 08h30
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @OU-NATIONAL (National organizatio...)
                    </td>
                    <td>
                      message 3
                    </td>
                  </tr>
                  <tr
                    class="HistoryLine"
                  >
                    <td
                      class="time"
                    >
                      4/27/ 16h27
                    </td>
                    <td
                      class="recipients"
                    >
                      <b>
                        To: 
                      </b>
                      @simpsons2020x00 (Survey on the Simpso...)
                    </td>
                    <td>
                      message de test
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="tableOptionsWrapper"
            >
              <ul
                class="paginationNav pagination pagination-sm"
                data-testid="pagination-nav"
              >
                <li
                  class="page-item active"
                >
                  <span
                    class="page-link"
                  >
                    1
                    <span
                      class="sr-only"
                    >
                      (current)
                    </span>
                  </span>
                </li>
                <li
                  class="page-item"
                >
                  <a
                    class="page-link"
                    href="#"
                    role="button"
                  >
                    2
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="textwidth"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
